src/instance.rs

```diff
@@ -214,6 +225,123 @@ unsafe impl<T> AsPyPointer for Py2<'_, T> {
     }
 }
 
+/// A borrowed equivalent to `Py2`.
+///
+/// The advantage of this over `&Py2` is that it avoids the need to have a pointer-to-pointer, as Py2
+/// is already a pointer to a PyObject.
+#[repr(transparent)]
+pub(crate) struct Py2Borrowed<'a, 'py, T>(NonNull<ffi::PyObject>, PhantomData<&'a Py2<'py, T>>); // TODO is it useful to have the generic form?
```

Comment:
So what is the verdict on the `TODO`? Is there already a use case for that? Otherwise, I'd suggest leaving it non-generic for now.

Reply:
I think #3651 might be the first use case, if we do actually think the thing I just pushed there has value.

Reply:
Understood, but then I'd suggest removing the `TODO` comment now (as the whole PR is in anticipation of future uses).